### PR TITLE
fix repeat adding parameters to the URL

### PR DESCRIPTION
**Pull request by BurningCN**
## What is the purpose of the change

ConfigParser#toParameterString and ConfigParser#toParameterString#appItemToUrls ，when after the toParameterString method call, the category parameter is added again

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).
image

### CODE REVIEW

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java [hunk 1]
```diff
@@ -137,8 +137,6 @@
 
     private static String toParameterString(ConfigItem item) {
         StringBuilder sb = new StringBuilder();
-        sb.append("category=");
-        sb.append(DYNAMIC_CONFIGURATORS_CATEGORY);
```
Comment (chickenlj) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java:141]: I think we should keep this two lines and remove `urlBuilder.append("&category=").append(DYNAMIC_CONFIGURATORS_CATEGORY);` 
in the caller mehod.

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java (hunk 1 above)
Comment (BurningCN) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java:141]: Yes, you are right. it's my fault that I didn't look carefully. method appItemToUrls also calls the method toParameterString, and it is filled with the "appdynamicconfigurators" parameter.Indeed, only the serviceItemToUrls method has the "category" parameter repeatedly added.

**Review by zonghaishang — APPROVED** [review]
LGTM

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java (hunk 1 above)
Comment (chickenlj) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java:141]: Thanks. All look fine now.

**Review by chickenlj — APPROVED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java (hunk 1 above)
Comment (chickenlj) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java:141]: Hi, do you have time to fix this issue so we can merge this patch soon？

**Review by chickenlj — CHANGES_REQUESTED** [review]
Please fix the review comment i left above

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java (hunk 1 above)
Comment (BurningCN) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/parser/ConfigParser.java]: I have modified it, but there are multiple commit logs under this PR, do I need to re-issue a new PR request to merge?

**Review by xiaoheng1 — APPROVED** [review]
LGTM

**Review by AlbumenJ — APPROVED** [review]
LGTM.

**Pull request merged**